swan-lake/learn-the-platform/ballerina-persist/overview.md

```diff
@@ -0,0 +1,67 @@
+redirect_from:
+- /learn/ballerina-persist/persist-overview/
+---
+This feature has three main sections: Persist Model, Persist CLI and Persist Client API. The Persist Model is used to define the data model. The Persist CLI is used to generate the client API for the data model. The Persist Client API is used to access the data in the data store.
+
+## Persist Model
+
+The Persist Model is used to define the data model. The data model is defined in a Ballerina source file inside the persist directory in the project root directory. The data model is defined using the Ballerina record type. The following example shows how to define a data model for a table named `Employee` with the fields `id`, `name`, `age` and `salary`.
+
+```ballerina
+type Employee record {
+    readonly int id;
+    string name;
+    int age;
+    float salary;
+}
+```
+
+The `readonly` keyword is used to define a field as a primary key. You need to have at least one primary key field in the Entity record. The primary key field is used to uniquely identify a record in the table. The primary key field is used to generate the `get`, `update`, and `delete` operations in the client API. You can define multiple primary key fields in the Entity record.
+
+Learn about how to define relationships between entity records and more about the persist model in the [Persist Model](/learn/persist-model/) section.
+
+> Note: The Ballerina VSCode plugin provides validation and code actions for the persist model. So you can easily create the persist model using the Ballerina VSCode plugin.
+
+## Persist CLI
+
+The `bal persist` CLI is used to generate the client API for the data model. Based on the data store, additional configurable file and setup scripts are generated. For example, if you are using MySQL as the data store, the `persist_db_config.bal` file, and the `script.sql` script are generated. The `persist_db_config.bal` file is used to configure the MySQL database connection. The `script.sql` file is used to create the table in the MySQL database. The client API is generated in the `generated` directory in the project root directory.
+
+The `bal persist` has two main built-in CLI commands: `persist init` and `persist generate`. The `persist init` command is used to initialize bal persist in the Ballerina project. The `persist generate` command is used to generate the client API for the data model. Additionally, there is an experimental `persist migrate` command to generate the SQL scripts for changing the table structure in the database when the data model is changed.
```

Comment:
```suggestion
The `bal persist` has two main built-in CLI commands: `persist init` and `persist generate`. The `persist init` command is used to initialize `bal persist` in the Ballerina project. The `persist generate` command is used to generate the client API for the data model. Additionally, there is an experimental `persist migrate` command to generate the SQL scripts for changing the table structure in the database when the data model is changed.
```
This paragraph is too lengthy. Shall we consider breaking them into multiple or summarizing them? Else, we can add a list.